packages/core/src/transaction.ts

```diff
@@ -226,8 +226,8 @@ class Transaction {
    * @param {object} meta The meta with bookmark
    * @returns {void}
    */
-  _onCompleteCallback(meta: { bookmark?: string | string[] }): void {
-    this._onBookmark(new Bookmark(meta.bookmark))
+  _onCompleteCallback(meta: { bookmark?: string }): void {
+    this._onBookmark(meta.bookmark)
```

Comment:
Why change `onBookmark` signature?

Reply:
Because I want to fail fast should the server send multiple bookmarks. According to the protocol spec, there should only ever be 0 or 1 bookmark in a `SUCCESS` message.

Reply:
Also, thinking further down the call stack, I want `_bookmarkOut` to be a string. How do you turn a `Bookmark` (which can contain a string array) into a single string in a type safe way?